### PR TITLE
#159521563 Enable selecting number of rows per page

### PR DESCRIPTION
**Pull request by kimotho-njoki**
## What does this PR do?
Enables selecting the number of rows to be displayed per page.

## Description of Task to be completed?
- Refactor relevant components to enable functionality in row selection
- Add relevant tests

## How should this be manually tested?
- On each table, click on the row dropdown on the footer and select the number of rows you would like displayed

## What are the relevant pivotal tracker stories?
[#159521563](https://www.pivotaltracker.com/n/projects/2146417/stories/159521563)

## Any background context you want to add?
N/A

## Important notes
N/A

## Packages installed
N/A

## Deployment note
N/A

## Related PRs branch | PR (branch_name) | (pr_link)
Branch | PR
------ | ------
other_pr_production | [link]()
other_pr_master | [link]()
## Todos
- [x] Raise PR
- [ ] Test
- [ ] Documentation
## Screenshots
N/A

### CODE REVIEW

**Comment by sachungo** [comment]
@kimotho-njoki please fix the conflicts. Also, we are using SASS so delete the `LoginComponent.css` if you still have it in your branch. It had been deleted in develop, unless someone re-introduce it again

**Comment by Arthur236** [comment]
Hi @sachungo, all the conflicts should be fixed now

File: src/_css/sharedStyling.scss [hunk 1]
```diff
@@ -217,7 +217,8 @@ input,
 }
 
 #page-limit {
-  width: 45%;
+  width: 75%;
```
Comment (sachungo) [inline src/_css/sharedStyling.scss]: I hope you have confirmed this won't break any other existing UI

File: src/components/AssetModels/AssetModelsComponent.jsx [hunk 2]
```diff
@@ -86,7 +91,8 @@ export class AssetModelsComponent extends React.Component {
 
             <Table.Footer>
               <Table.Row>
-                {!_.isEmpty(this.props.assetModels) && (
+                {!_.isEmpty(this.props.assetModels) &&
+                (
```
Comment (sachungo) [inline src/components/AssetModels/AssetModelsComponent.jsx]: Nitpick: Move the opening brace to the previous line, the way it was before the change

File: src/components/User/UserComponent.jsx [hunk 3]
```diff
@@ -65,48 +67,52 @@ const UserDetailsComponent = (props) => {
...
-                      id="page-limit"
-                      placeHolder="Show Rows"
-                      options={rowOptions}
-                      upward
-                    />
-                  </Segment>
-                </Segment.Group>
-              </Table.HeaderCell>
-            )}
+            <Table.HeaderCell colSpan="4" id="pagination-header">
```
Comment (sachungo) [inline src/components/User/UserComponent.jsx]: The whole change to `<Table.HeaderCell />` is unnecessary and it messes the indentation. Please revert to how it was before and only add the needed props of `value` and `onChange` to `DropdownComponent`

File: src/components/common/DropdownComponent.jsx [hunk 4]
```diff
@@ -5,7 +5,6 @@ import PropTypes from 'prop-types';
 const DropdownComponent = props => (
   <Dropdown
     id={props.id}
-    style={{ width: '65%' }}
```
Comment (sachungo) [inline src/components/common/DropdownComponent.jsx]: This deletion won't affect anything, right?

File: src/_css/sharedStyling.scss (hunk 1 above)
Comment (Arthur236) [reply src/_css/sharedStyling.scss]: Yes. The `page-limit` id is only being used on the `DropdownComponent`

File: src/components/common/DropdownComponent.jsx (hunk 4 above)
Comment (Arthur236) [reply src/components/common/DropdownComponent.jsx]: No, it won't. I added the styling for it in the `sharedStyling.scss` file

**Comment by Arthur236** [comment]
Regarding the dashboard. You're using the old version of the dashboard. This is what the dashboard should look like
dashboard

**Comment by sachungo** [comment]
Ohh my...I just checked out your branch locally. I guess it's time to clear my cache

**Comment by kimotho-njoki** [comment]
@sachungo 👍

**Pull request merged**